### ISSUE TITLE
Passing a DB to the constructor don't work

### ISSUE THREAD

**Issue by pabloalmunia** [mafintosh/mongojs#242]
With the current version of mongojs and mongodb, this code dont' work:

```
var mongodb = require('mongodb');
var mongojs = require('mongojs');
mongodb.connect('mongodb://localhost/test', function (err, theDb) {
    var db = mongojs(theDb, ['myCollection']);
});
```

This is the error: **"You must pass a connection string or a mongojs instance."**

I'm thinking that this is a error of *to-mongodb-core*, because the error is a problem with this two lines into to-mongodb-core/index.js:

```
  if (typeof db.getCollectionNames !== 'function') return false;
  if (typeof db._getServer !== 'function') return false;
```

**Comment by mgmorcos** [comment]
:+1:

**Comment by watson** [comment]
I just checked the current version of the [mongodb](https://github.com/mongodb/node-mongodb-native) module (v2.0.47). Its `.connect()` function calls the callback with an object that looks like this:

```js
{
  domain: null,
  _events: {},
  _maxListeners: 10,
  s: { ... },
  serverConfig: [Getter],
  bufferMaxEntries: [Getter],
  databaseName: [Getter],
  options: [Getter],
  native_parser: [Getter],
  slaveOk: [Getter],
  writeConcern: [Getter]
}
```

The [to-mongodb-core](https://github.com/sorribas/to-mongodb-core) module [checks](https://github.com/sorribas/to-mongodb-core/blob/6f575b1bdddcf27bf6a845464b454823e9e9285e/index.js#L2-L5) for the presence of:

- `.close`
- `.collection`
- `.getCollectionNames`
- `._getServer`

Neither of which exists on the current version of the db object.

After doing a little more digging [it seems](https://github.com/sorribas/to-mongodb-core/blob/6f575b1bdddcf27bf6a845464b454823e9e9285e/test.js#L8-L10) that the to-mongodb-core module actually expects a reference to the mongojs module and **not** the mongodb module. In that case the README.md file is wrong in suggesting that.

@sorribas What was your thoughts on this functionality?

**Comment by saintedlama** [comment]
Should be fixed in version 2.0 (released just some seconds ago) - we accept a mongodb-native connections now

**Comment by saintedlama** [comment]
Any update?
